src/common/ServerCompatBanner.js

```diff
@@ -0,0 +1,101 @@
+import { useSelector, useDispatch } from '../react-redux';
+import Label from './Label';
+import { getActiveAccount } from '../account/accountsSelectors';
+import { getIsAdmin, getSession } from '../directSelectors';
+import { dismissCompatNotice } from '../session/sessionActions';
+import ZulipTextButton from './ZulipTextButton';
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+const styles = createStyleSheet({
+  wrapper: {
+    backgroundColor: HALF_COLOR,
+    padding: 8,
+  },
+  textRow: {
+    padding: 8,
+    flexDirection: 'row',
+  },
+  buttonsRow: {
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+});
+
+type Props = $ReadOnly<{||}>;
+
+/**
+ * A "nag banner" saying that the server version is unsupported.
+ *
+ * Currently just checks if it's less than 2.1.0, since that's the
+ * marker that was put down in zulip/zulip@d7d79b57b. In the future,
+ * this won't have to be so hard-coded; we may use a timer or
+ * something.
+ */
+// Made with somewhat careful attention to
+// https://material.io/components/banners. Please consult that before
+// making layout changes, and try to make them in a direction that
+// brings us closer to those guidelines.
+export default function ServerCompatBanner(props: Props) {
+  const dispatch = useDispatch();
+  const hasDismissedServerCompatNotice = useSelector(
+    state => getSession(state).hasDismissedServerCompatNotice,
+  );
+  const zulipVersion = useSelector(state => getActiveAccount(state).zulipVersion);
+  const realm = useSelector(state => getActiveAccount(state).realm);
+  const isAdmin = useSelector(getIsAdmin);
+
+  if (zulipVersion && !zulipVersion.isAtLeast('2.1.0') && !hasDismissedServerCompatNotice) {
+    return (
+      <SafeAreaView mode="padding" edges={['right', 'left']} style={styles.wrapper}>
+        <View style={styles.textRow}>
+          {isAdmin ? (
+            <Label
+              text={{
+                text:
+                  '{realm} is running Zulip Server {serverVersionRaw}, which is unsupported. Please upgrade your server as soon as possible.',
+                values: { realm: realm.toString(), serverVersionRaw: zulipVersion.raw() },
+              }}
+            />
+          ) : (
+            <Label
+              text={{
+                text:
+                  '{realm} is running Zulip Server {serverVersionRaw}, which is unsupported. Please contact your administrator about upgrading.',
+                values: { realm: realm.toString(), serverVersionRaw: zulipVersion.raw() },
+              }}
+            />
+          )}
+        </View>
+        <View style={styles.buttonsRow}>
+          <ZulipTextButton
+            leftMargin
+            label="DISMISS"
+            onPress={() => {
+              dispatch(dismissCompatNotice());
+            }}
+          />
+          <ZulipTextButton
+            leftMargin
+            label={isAdmin ? 'FIX NOW' : 'LEARN MORE'}
```

Comment:
If we can, I think it'd be better to have the string begin as "Learn more", and use styling to transform it to all-caps. Among other things that seems less confusing for translators.

That's in fact what the canonical Material implementation for the web seems to do:
https://material.io/components/buttons/web#text-button
![image](https://user-images.githubusercontent.com/28173/119580439-42888a80-bd75-11eb-8a78-e31b12cbbc6e.png)

And it looks like RN's text styles offer the necessary feature for that:
https://reactnative.dev/docs/text-style-props#texttransform
i.e `textTransform: 'uppercase'`.

(Which means this aspect should move into the `ZulipTextButton` implementation.)

Reply:
Ah OK, got it! I presume `textTransform: 'uppercase'` wouldn't error on languages that don't have an 'uppercase' style; that seems like something that they would have thought of.

Reply:
Yeah, agreed.